src/htm/engine/RegionImpl.cpp

```diff
@@ -45,21 +44,17 @@ std::string RegionImpl::getName() const { return region_->getName(); }
 // templated methods can't be virtual and thus can't be
 // overridden by subclasses.
 
-#define getParameterInternalT(MethodT, Type)                                   \
-  Type RegionImpl::getParameter##MethodT(const std::string &name,              \
-                                         Int64 index) {                        \
-    if (!region_->getSpec()->parameters.contains(name))                        \
-      NTA_THROW << "getParameter" #Type ": Region type " << getType()          \
-	            << ", parameter " << name                                      \
-                << " does not exist in region spec";                              \
-    ParameterSpec p = region_->getSpec()->parameters.getByName(name);          \
-    if (p.dataType != NTA_BasicType_##MethodT)                                 \
-      NTA_THROW << "getParameter" #Type ": Region type " << getType()          \
-	            << ", parameter " << name                                      \
-                << " is of type " << BasicType::getName(p.dataType)            \
-                << " not " #Type;                                              \
-    NTA_THROW << "getParameter" #Type " --  parameter '"       \
-                << name << "' on region of type " << getType() << " not implemented"; \
+#define getParameterInternalT(MethodT, Type)                                                                           \
```

Comment:
nit, not part of this PR, but could this be a real method, not `define`d?

Reply:
>  but could this be a real method, not defined?

No, I tried every which way and could not make it work as a templated function.  Passing the method as a function pointer we can do but passing the type is hard.

Reply:
Actually I have thought about how to re-engineer how arguments are get/set in the region after the region is created.  I just did not want to break the API.

Reply:
> Actually I have thought about how to re-engineer how arguments are get/set in the region after the region is created. I just did not want to break the API. 

maybe we can start thinking about NAPIv2 (along with the cleanup of obsoleted method + providing the RESTful interface) .